tests/unit/controlgroup/core.js

```diff
@@ -160,4 +160,39 @@ QUnit.test( "Single controlgroup button - vertical", function( assert ) {
 		" ui-corner-tr ui-corner-tl ui-corner-bl ui corner-br" );
 } );
 
+QUnit.module( "Controlgroup: Non-empty class key", {
+	setup: function() {
+		this.classKey = $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ];
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] =
+			$.trim( $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] +
+				" something-custom" );
+	},
+	teardown: function() {
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] = this.classKey;
+	}
+} );
+
+QUnit.test( "Controlgroup handles non-empty class key correctly", function( assert ) {
+	assert.expect( 2 );
+
+	$( ".controlgroup-non-empty-class-key" )
+		.controlgroup()
+		.controlgroup( "instance" );
+
+	assert.deepEqual(
```

Comment:
So, the user innocently sets ```$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] to, say, "ui-shadow"``` and we proceed to mangle that value in every instance of a selectmenu that resides inside a controlgroup. Oh, and it's worse, because the length of the string increases with each ```refresh()```.